packages/resource-deployment/scripts/function-app-create.sh

```diff
@@ -28,27 +27,55 @@ while getopts "r:" option; do
     esac
 done
 
-if [ -z "$resourceGroupName" ]; then
+if [ -z $resourceGroupName ]; then
     exitWithUsageInfo
 fi
 
+# Will return the function name if we deployed before the function app template on this resource group
+functionAppName=$(az group deployment show -g "$resourceGroupName" -n "function-app-template"  --query "properties.parameters.name.value" -o tsv)
+
+# Boolean to track if we should create a new App Registration or there's one already existing
+createNewApp=true
+ 
+# Check if the Function App name is valid and the Function App exists
+if [ ! -z "$functionAppName" ] && (az functionapp show -n "$functionAppName" -g "$resourceGroupName" 1>/dev/null); then
+    # Get the Client (App) ID for the App Registration that's used for this Function App authentication
+    clientId=$(az webapp auth show -n "$functionAppName" -g "$resourceGroupName" --query "clientId" -o tsv)
+    # Check if the App Registration name is valid and the App Registration exists
+    if [ ! -z "$clientId" ] && (az ad app show --id "$clientId" 1>/dev/null); then
+        # If we hit this line, that means we don't need to create a new App Registration
+        createNewApp=false
+    fi
+fi
+
+if ("$createNewApp" = true); then
```

Comment:
Can make this part a function ``createAppRegistration()``? 
Then we don't have to deal with the boolean value which seems not supported in bash scripts?

Reply:
If we create a function, we'll move same logic to the function which still has boolean, unless you want to Not all the conditions in the if's and create assume that the default is we don't need to create a new one, which I don't like to do, I'd like to use the boolean as a guard for the knwon case that we don't need to create a new app

Reply:
you can still move it under createAppRegistration